src/main/java/org/snomed/snowstorm/core/data/services/AutoMerger.java

```diff
@@ -0,0 +1,518 @@
+                reapplyDescriptionChanges(mapByIdentifier(sourceConcept.getDescriptions()), mapByIdentifier(targetConceptOld.getDescriptions()), mapByIdentifier(targetConceptNew.getDescriptions()), changedComponentIds, mergedConcept);
+            } else if (ComponentType.Relationship.equals(changedComponentType)) {
+                reapplyRelationshipChanges(mapByIdentifier(sourceConcept.getRelationships()), mapByIdentifier(targetConceptOld.getRelationships()), mapByIdentifier(targetConceptNew.getRelationships()), changedComponentIds, mergedConcept);
+            } else if (ComponentType.ClassAxiom.equals(changedComponentType)) {
+                mergedConcept.setClassAxioms(reapplyAxiomsChanges(mapAxiomByIdentifier(sourceConcept.getClassAxioms()), mapAxiomByIdentifier(targetConceptOld.getClassAxioms()), mapAxiomByIdentifier(targetConceptNew.getClassAxioms()), changedComponentIds, mergedConcept));
+            } else if (ComponentType.GciAxiom.equals(changedComponentType)) {
+                mergedConcept.setGciAxioms(reapplyAxiomsChanges(mapAxiomByIdentifier(sourceConcept.getGciAxioms()), mapAxiomByIdentifier(targetConceptOld.getGciAxioms()), mapAxiomByIdentifier(targetConceptNew.getGciAxioms()), changedComponentIds, mergedConcept));
+            } else if (ComponentType.LanguageReferenceSetMember.equals(changedComponentType)) {
+                reapplyLangMemberChanges(mapLangMembersByIdentifier(sourceConcept.getDescriptions()), mapLangMembersByIdentifier(targetConceptOld.getDescriptions()), mapLangMembersByIdentifier(targetConceptNew.getDescriptions()), changedComponentIds, mergedConcept);
+            }
+        }
+
+        return mergedConcept;
+    }
+
+    private void reapplyConceptChanges(Concept sourceConcept, Concept targetConceptNew, Concept targetConceptOld, Set<String> value, Concept mergedConcept) {
+        mergedConcept.setActive(getValueChanged(targetConceptOld.isActive(), targetConceptNew.isActive(), sourceConcept.isActive()));
+        mergedConcept.setModuleId(getValueChanged(targetConceptOld.getModuleId(), targetConceptNew.getModuleId(), sourceConcept.getModuleId()));
+        mergedConcept.setDefinitionStatus(getValueChanged(targetConceptOld.getDefinitionStatus(), targetConceptNew.getDefinitionStatus(), sourceConcept.getDefinitionStatus()));
+        mergedConcept.setReleaseHash(getValueChanged(targetConceptOld.getReleaseHash(), targetConceptNew.getReleaseHash(), sourceConcept.getReleaseHash()));
+        mergedConcept.setReleased(getValueChanged(targetConceptOld.isReleased(), targetConceptNew.isReleased(), sourceConcept.isReleased()));
+
+        mergedConcept.updateEffectiveTime();
+
+        // If changed, ensure component is re-activated unless the RHS had the inactivation.
+        if (mergedConcept.getEffectiveTime() == null && !sourceConcept.isActive() && targetConceptNew.isActive()) {
+            mergedConcept.setActive(true);
+        }
```

Comment:
Without this block (and the others like it), the function basically takes the Source concept and what has changed on the Target concept and combines the two. As a result, we could end up with the middle column having a different module but also remaining inactive. This block re-activates the Concept (and other components in their respective methods) if there has been a change.